src/Runtime/ISqlCommand.fs

```diff
@@ -188,23 +188,18 @@ type ``ISqlCommand Implementation``(cfg: DesignTimeConfig, connection, commandTi
         let hasOutputParameters = cmd.Parameters |> Seq.cast<NpgsqlParameter> |> Seq.exists (fun x -> x.Direction.HasFlag( ParameterDirection.Output))
 
         if not hasOutputParameters
-        then 
-            let xs = Seq.delay <| fun() -> 
-                ``ISqlCommand Implementation``
-                    .ExecuteReader(cmd, setupConnection, readerBehavior, parameters, expectedColumns)
-                    .MapRowValues<'TItem>( rowMapper)
+        then
+            use reader = ``ISqlCommand Implementation``.ExecuteReader(cmd, setupConnection, readerBehavior, parameters, expectedColumns)
+            let xs = reader.MapRowValues<'TItem>(rowMapper) |> Seq.toList :> seq<'TItem>
```

Comment:
enumeration of sequence is required within scope where reader is not disposed yet.

Reply:
This makes sense.  The former version will implicitly create the reader object, but then its lifetime is uncertain (as dsyme likes to say, leaving GC to clean up something is like expecting your teenager to clean the floor of their room - it will happen eventually).   Since the reader hangs on to underlying connection, leaving reader opens threatens connection pool reuse.

Reply:
I can see @dmitry-a-morozov 's concern though - this whole construct generating a sequence is creating something that can only be processed once.  If you consume the sequence twice it will probably not work.  This is a catch 22 - if you leave the whole construct live,  your underlying resource usage (connection) can linger,  but if you clean it up promptly, you can't use the seq<>  twice.